modules/MusicDownloader.js

```diff
@@ -38,19 +44,45 @@ async function downloader(url) {
     }
 }
 
-async function downloadSong(url) {
+async function downloadSong(url, votes) {
     const spotify = new Spotify({
         clientId: process.env.SPOTIFY_CLIENT_ID,
         clientSecret: process.env.SPOTIFY_CLIENT_SECRET,
     });
 
-    const data = await spotify.getTrack(url);
-    const file = sterylizator(data.artists.join('-')+'_'+data.name);
-    logger('log',`Pobieram: ${data.name+' by: '+data.artists.join(', ')}`,'downloadSong');
-    if (fs.existsSync(`./mp3/onDemand/${file}.mp3`)) return logger('warn',`Plik istnieje!`,'downloadSong');
-    const song = await spotify.downloadTrack(url);
-    fs.writeFileSync(`./mp3/onDemand/${file}.mp3`, song);
-    return logger('log','Pobrano :>','downloadSong');
+    try {
+        const data = await spotify.getTrack(url);
+        const file = sterylizator(data.artists.join('-') + '_' + data.name);
+        logger('log', `Pobieram: ${data.name + ' by: ' + data.artists.join(', ')}`, 'downloadSong');
+        logger('verbose', 'Sprawdzanie czy nie pobieram z głosów...', 'downloadSong');
+        if (votes) {
+            logger('verbose', 'Pobieranie z głosowania!', 'downloadSong');
+            if (fs.existsSync(`./mp3/7/${file}.mp3`)) return logger('warn', `Plik istnieje!`, 'downloadSong');
+            const song = await spotify.downloadTrack(url);
+            fs.writeFileSync(`./mp3/7/${file}.mp3`, song);
+            logger('verbose', 'Normalizacja dźwięku przy użyciu mp3gain...', 'downloadSong');
+            exec(`mp3gain -r -c ./mp3/7/${file}.mp3`, (error, stdout, stderr) => {
+                logger('verbose', "\n"+ stdout, 'downloadSong')
+                if (global.debugmode === true) {
+                    DebugSaveToFile('MusicDownloader', 'downloadSong', 'mp3gain', stdout);
+                    logger('verbose', `Zapisano do debug/`, 'downloadSong');
+                }
+            });
+            return logger('log', 'Pobrano :>', 'downloadSong');
+        }
+        if (fs.existsSync(`./mp3/onDemand/${file}.mp3`)) return logger('warn', `Plik istnieje!`, 'downloadSong');
+        const song = await spotify.downloadTrack(url);
+        fs.writeFileSync(`./mp3/onDemand/${file}.mp3`, song);
+        exec(`mp3gain -r -c ./mp3/onDemand/${file}.mp3`, (error, stdout, stderr) => logger('verbose',"\n" + stdout, 'downloadSong'));
+        return logger('log', 'Pobrano :>', 'downloadSong');
+    } catch (e) {
+        logger('error', "Błąd w trakcie wykonywania funkcji downloadSong", 'downloadSong');
+        logger('error', e, 'downloadSong');
+        if (global.debugmode === true) {
+            DebugSaveToFile('MusicDownloader', 'downloadSong', 'error_main_function', e);
```

Comment:
Zmień nazwę error_main_function na catched_error
```suggestion
            DebugSaveToFile('MusicDownloader', 'downloadSong', 'catched_error', e);
```